### PR TITLE
Dex walkthrough

### PR DESCRIPTION
**Pull request by psorinionut**
#### Did you test the changes locally ?
- [x] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [ ] document new feature
- [ ] update documentation that is not relevant anymore
- [x] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

### CODE REVIEW

File: docs/developers/tutorials/dex-walkthrough.md [hunk 1]
```diff
@@ -0,0 +1,920 @@
...
+
+[comment]: # (mx-context-auto)
+
+## Prerequisites
+
+The DEX contracts are a bit more advance than you standard SCs, so basic knowledge about Rust SC development is required. If you are a beginner, an easier starting point (like the Crowdfunding SC or the Staking SC tutorials) is strongly advised. Also, to better grasp the DEX contracts implementation, it is important that you first understand the xExchange economic model.
```
Comment (claudiulataretu) [inline docs/developers/tutorials/dex-walkthrough.md:16]: `than a standard SC`

File: docs/developers/tutorials/dex-walkthrough.md [hunk 2]
```diff
@@ -0,0 +1,920 @@
...
+You can find the xExchange whitepaper here:
+https://xexchange.com/x-exchange-economics.pdf
+:::
+
+[comment]: # (mx-context-auto)
+
+## **DEX repo structure** & recommendations
+
+The main DEX contracts are as follow:
+
```
Comment (claudiulataretu) [inline docs/developers/tutorials/dex-walkthrough.md:28]: missing router contract

Comment (psorinionut) [reply docs/developers/tutorials/dex-walkthrough.md:28]: This documentation was written with a developer go to summary approach in mind, so I did not cover all the contracts, but only those that are usually used by external developers that are trying to build on top of the DEX. For that reason, I skipped the router contract, as that one is more of a internal utility contract. That being said, I've just realised that the Router contract also has the `multiPairSwap` endpoint which can indeed be used by other developers. I'll come back with a walkthrough for that endpoint as well.

File: docs/developers/tutorials/dex-walkthrough.md (hunk 1 above)
Comment (psorinionut) [reply docs/developers/tutorials/dex-walkthrough.md]: Updated.

File: docs/developers/tutorials/dex-walkthrough.md (hunk 2 above)
Comment (psorinionut) [reply docs/developers/tutorials/dex-walkthrough.md:28]: Updated.

**Review by claudiulataretu — APPROVED** [review]

**Pull request merged**